docs/ai/conceptual/prompt-engineering-in-dot-net.md

```diff
@@ -0,0 +1,189 @@
+First president to be elected to a second term in office.
+First president to receive votes from every presidential elector in an election.
+First president to fill the entire body of the United States federal judges; including the Supreme Court.
+First president to be declared an honorary citizen of a foreign country, and an honorary citizen of France.
+John Adams ...  // Text truncated
+// Cues
+DOMESTIC POLICY
+- George Washington: 
+- John Adams:";
+```
+
+- **DOMESTIC POLICY** shows the model that you want it to start each group with the category in all caps.
+- **- George Washington:** shows the model to start each section with George Washington's accomplishments listed on one line.
+- **- John Adams:** shows the model that it should list remaining presidents in chronological order.
+
+## .NET implementations
+
+This section compares two .NET options for prompt engineering: [Semantic Kernel](/semantic-kernel/overview/) and [Azure OpenAI](/azure/ai-services/openai/overview).
+
+### Semantic Kernel
+
+Semantic Kernel is an SDK that integrates Large Language Models (LLMs) with conventional programming languages such as C#, Python, and Java. Semantic Kernel lets you define plugins and easily chain them together in just a few lines of code. You can use [Semantic Kernel planners](/semantic-kernel/agents/planners/?tabs=Csharp) to automatically orchestrate plugins, which lets you ask an LLM to generate a plan that Semantic Kernel can execute to achieve a user's unique goal.
+
+By deeply integrating with Visual Studio Code, Semantic Kernel also makes it easy for you to integrate [prompt engineering](/semantic-kernel/prompts/your-first-prompt?tabs=Csharp) into your existing development processes:
+
+- Create prompts directly in your preferred code editor.
+- Write tests for your prompts using your existing testing frameworks.
+- Deploy your prompts to production using your existing CI/CD pipelines.
+
+SK Completions use GPT-3 and GPT-3.5 models, which have no specific format rules for prompts. The preceding sections all contain examples of SK Completions.
+
+SK Chat Completions use GPT-35-Turbo and GPT-4 models, which use a specific chat-like format consisting of role-based messages as shown in the following example.
+
+```csharp
+prompt = @$"<message role=""system"">You are helping students with US History homework. Answers should not be about other countries except in relation to the US. Provide some supporting information for answers. Format any summaries using lists.</message>
+<message role=""user"">Which president had the shortest term?<message/>
+<message role=""assistant"">William Henry Harrison died just 31 days after his inauguration as president in 1841, making his presidency the shortest in U.S. history.<message/>
+<message role=""user"">Instructions: Summarize US Presidential accomplishments, grouped by category. Categories: Domestic Policy, US Economy, Foreign Affairs, Space Exploration, Other.</message>";
+```
```

Comment:
Here, raw string literals will look much cleaner:

```suggestion
```csharp
prompt = """
<message role="system">
    You are helping students with US History homework. Answers should not be about other countries except in 
    relation to the US. Provide some supporting information for answers. Format any summaries using lists.
</message>
<message role="user">Which president had the shortest term?<message/>
<message role="assistant">
    William Henry Harrison died just 31 days after his inauguration as president in 1841,
    making his presidency the shortest in U.S. history.
<message/>
<message role="user">
    Instructions: Summarize US Presidential accomplishments, grouped by category.
    Categories: Domestic Policy, US Economy, Foreign Affairs, Space Exploration, Other.
</message>
""";
```
```